### PR TITLE
Add tasks to Gitpod

### PR DESCRIPTION
**Pull request by edmundmiller**
Adds a step to pre-install dependencies, which will get caught in the pre-build.

[GitPod Docs](https://www.gitpod.io/docs/configure/workspaces/tasks)

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -32,3 +32,8 @@ github:
     addComment: false
     # add a "Review in Gitpod" button to the pull request's description (defaults to false)
     addBadge: true
+tasks:
+  - name: Dev Server main-site
+    init: npm install
+    command: npm run dev --workspace sites/main-site
```
Comment (mashehu) [inline .gitpod.yml:38]:
```suggestion
```
It usually is another subsite on git pod because you get main-site from the netlify deploy

Comment (mashehu) [reply .gitpod.yml:38]: And move it maybe to the top where the commented out task is

**Review by mashehu — APPROVED** [review]

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
@mashehu if you happy and merge in I can test on your pytest migration PR

**Comment by mashehu** [comment]
sorry, not happy yet. better to do the manual step for now, so we can get the docs updated, @jfy133  🙂

File: .gitpod.yml [hunk 2]
```diff
@@ -1,34 +1,16 @@
...
-    pullRequestsFromForks: true
-    # add a check to pull requests (defaults to true)
-    addCheck: true
-    # add a "Review in Gitpod" button as a comment to pull requests (defaults to false)
-    addComment: false
-    # add a "Review in Gitpod" button to the pull request's description (defaults to false)
-    addBadge: true
+tasks:
+  # NOTE adding multiple tasks causes them to all run
+  - name: Dev Server main-site
```
Comment (mashehu) [inline .gitpod.yml:14]:
```suggestion
  - name: install dependencies and extract cache
```

**Comment by mashehu** [comment]
@nf-core-bot fix linting

**Pull request merged**